airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_stream_config.py

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from dataclasses import dataclass
+from typing import Any, List, Mapping, Optional
+
+from airbyte_cdk.models import ConfiguredAirbyteCatalog
+from airbyte_cdk.sources.file_based.exceptions import ConfigValidationError
+from airbyte_cdk.sources.file_based.remote_file import FileType
+from airbyte_cdk.sources.file_based.schema_validation_policies import UserValidationPolicies
+
+
+@dataclass
+class FileBasedStreamConfig:
```

Comment:
I think we can get rid of the parsing code if we define this as a `BaseModel`, and rely on pydantic to parse . Is that desirable, or do we want to avoid pydantic in this context?

Reply:
I like it, updated!